[Hack n Hustle] Get and update common fields across advisories

thegreyd opened the pull request:
I wanted a command to update the shipping date of a group of advisories without having to go through ET.

Made it a little general purpose so you can get any field across advisories, if you wanted to.
And then update common fields across advisories like date, manager, qe, owner etc. 
I don't see us updating the others as much
and date as basically the only real feature :) 
but worth it

vfreex commented on elliottlib/cli/advisory_commons_cli.py:
```diff
@@ -0,0 +1,103 @@
+from __future__ import absolute_import, print_function, unicode_literals
```
Python 3 doesn't need this.

vfreex commented on elliottlib/cli/advisory_commons_cli.py:
```diff
@@ -0,0 +1,103 @@
+@cli.command("advisory-commons", short_help="Show or update advisory commons for a group of advisories")
+@click.option("--advisories", metavar='ADVISORY',
+              help="Select given advisory id(s). comma separated")
+@click.option("--field", required=True,
+              help="Advisory field (like publish_date) to show or update. Examples of show only fields: ["
+                   "'errata_state', "
+                   "'errata_name', 'content_types', 'synopsis', "
+                   "'errata_bugs', "
+                   "'errata_type']")
+@click.option("--new",
+              help=f"New value of field to update advisories with. --field should be one of"
+                   f" {supported_update_fields}")
```
Would like to have separate options for each supported field for better validation, help message, and being able to update multiple fields.

thegreyd replied:
Maybe next hack n hustle :sweat_smile: ?
Don't want to spend too much time on this right now.. date format is validated in errata_tool which I think is sufficient for the initial implementation

thegreyd replied:
Added a --version-replace option specifically for topic, synopsis and description

vfreex commented on elliottlib/cli/advisory_commons_cli.py:
```diff
@@ -0,0 +1,103 @@
+
+
+@cli.command("advisory-commons", short_help="Show or update advisory commons for a group of advisories")
+@click.option("--advisories", metavar='ADVISORY',
+              help="Select given advisory id(s). comma separated")
```
Guess passing advisories as arguments in stead of options would be more ergonomics?
```suggestion
@click.argument("advisories", nargs=-1, type=click.IntRange(1), required=False)
```

thegreyd replied:
Yes! I didn't know argument could be required=False :) that is definitely better.

vfreex commented on elliottlib/cli/advisory_commons_cli.py:
```diff
@@ -0,0 +1,103 @@
+                        click.echo(f"Would have changed field {field} {current} ➔ {new}")
+                    else:
+                        update_kwarg = {}
+                        if field == 'publish_date':
+                            update_kwarg['date'] = new
```
I didn't see a compelling reason to rename this field. We have called this field `date` or `release_date` elsewhere in Elliott code.

thegreyd replied:
The weird thing is errata_tool's advisory.update() method searches for "date" specifically, any other field name is ignored.

vfreex commented on elliottlib/cli/advisory_commons_cli.py:
```diff
@@ -0,0 +1,103 @@
+
+    $ elliott -g openshift-4.8 --assembly 4.8.8 advisory-commons --field "publish_date" --new "2021-Aug-31" --noop
+"""
+    count_flags = sum(map(bool, [runtime.assembly != 'stream', advisories]))
+    if count_flags > 1:
+        raise click.BadParameter("Use only one of --assembly or --advisories")
```
For releases without assembly support (like 3.11), `runtime.assembly` is `None`.

BTW I don't think this check is absolutely needed.

thegreyd replied:
fixed!

vfreex reviewed (commented):
Thank you for adding this useful command in Hack'n'Hustle. Just some nits.

joepvd commented:
Nice! Would be great with a search-and-replace for the title in case, say, 4.8.8 gets renamed as 4.8.9.

thegreyd commented:
> Nice! Would be great with a search-and-replace for the title in case, say, 4.8.8 gets renamed as 4.8.9.
> 
> Apologies for the scope creep ;)

I added a --version-replace option as a shortcut to bump versions. PTAL
Noticed that we can't do a blanket search and replace in description (because of bug titles being part of it), so I added template lines

vfreex commented:
LGTM once you fix the conflict.

joepvd reviewed (approved):
/lgtm

Maybe squash or rebase into a single commit.
